includes/class-events.php

```diff
@@ -89,56 +91,41 @@ public function populate_concurrent_action_whitelist() {
 	}
 
 	/**
-	 * List events pending for the current period
+	 * List events pending for the current period.
 	 *
-	 * @param array $job_queue_size   Maximum number of events to return (excludes internal events).
-	 * @param array $job_queue_window How many seconds into the future events should be fetched.
-	 * @return array
+	 * @param null|int $job_queue_size   Maximum number of events to return (excludes internal events).
+	 * @param null|int $job_queue_window How many seconds into the future events should be fetched.
+	 * @return array Events to be run in the next batch.
 	 */
-	public function get_events( $job_queue_size = null, $job_queue_window = null ) {
+	public function get_events( $job_queue_size = null, $job_queue_window = null ): array {
 		$job_queue_size   = is_null( $job_queue_size ) ? JOB_QUEUE_SIZE : $job_queue_size;
 		$job_queue_window = is_null( $job_queue_window ) ? JOB_QUEUE_WINDOW_IN_SECONDS : $job_queue_window;
 
-		$events = get_option( 'cron' );
+		// Grab relevant events that are due, or soon will be.
+		$current_time = time();
+		$events = self::query( [
+			'timestamp' => [ 'from' => 0, 'to' => $current_time + $job_queue_window ],
+			'status'    => Events_Store::STATUS_PENDING,
+			'limit'     => -1, // Need to get all, to ensure we grab internals even when queue is backed up.
+		] );
```

Comment:
This is a pretty nice improvement :). Rather than querying ALL events, we can just grab the ones we are interested in (pending events within the time window).